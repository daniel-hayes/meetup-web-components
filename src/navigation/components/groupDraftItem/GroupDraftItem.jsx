import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
// import Icon from '../../../media/Icon';

export const PROFILE_GROUP_DRAFT_LIST_ITEM_CLASS = 'profileDropdown-draft-group';

export const GroupDraftItem = ({ groupDraft }) => {
	return (
		<li
			key={groupDraft.editLink}
			className={cx(PROFILE_GROUP_DRAFT_LIST_ITEM_CLASS, 'list-item')}
		>
			<p>{groupDraft.name}</p>
			<div>
				<span>{groupDraft.status}</span>
				<a className="draftprofiledropdown" href={groupDraft.editLink}>
					<strong>{groupDraft.actionTitle}</strong>{' '}
					{/* <Icon className="text--blue" shape="arrow-right" size="xs" /> */}
				</a>
			</div>
		</li>
	);
};

export default GroupDraftItem;

GroupDraftItem.propTypes = {
	groupDraft: PropTypes.shape({
		editLink: PropTypes.string,
		name: PropTypes.string,
		status: PropTypes.string,
		actionTitle: PropTypes.string,
	}),
};
