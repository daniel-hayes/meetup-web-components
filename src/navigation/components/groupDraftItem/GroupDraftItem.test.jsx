import React from 'react';
import { shallow } from 'enzyme';

import GroupDraftItem from './GroupDraftItem';

const MOCK_PROPS = {
	groupHome: jest.fn(),
	groupDraft: {
		editLink: 'create',
		name: 'Name of the group',
		status: 'In progress',
		actionTitle: 'Finish group',
	},
};

describe('Profile Dropdown', () => {
	const wrapper = shallow(<GroupDraftItem {...MOCK_PROPS} />);
	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should show the draft group at the top ProfileDropdown for authenticated screens', () => {
		const wrapperWithDrafts = shallow(<GroupDraftItem {...MOCK_PROPS} />);

		expect(wrapperWithDrafts.find('.profileDropdown-draft-group').length).toBe(1);
	});

	it('should has the class for the tracking integration', () => {
		const wrapperWithDrafts = shallow(<GroupDraftItem {...MOCK_PROPS} />);

		expect(wrapperWithDrafts.find('.draftprofiledropdown').length).toBe(1);
	});
});
