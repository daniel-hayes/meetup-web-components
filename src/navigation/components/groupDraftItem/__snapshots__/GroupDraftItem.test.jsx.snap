// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Dropdown should match snapshot 1`] = `
<li
  className="profileDropdown-draft-group list-item"
  key="create"
>
  <p>
    Name of the group
  </p>
  <div>
    <span>
      In progress
    </span>
    <a
      className="draftprofiledropdown"
      href="create"
    >
      <strong>
        Finish group
      </strong>
       
      <Icon
        className="text--blue"
        shape="arrow-right"
        size="xs"
      />
    </a>
  </div>
</li>
`;
